File: backend/src/AuctionLab.Domain/Entities/Auction.cs
namespace AuctionLab.Domain.Entities;

public class Auction
{
    public int AuctionId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal StartingPrice { get; set; }
    public decimal? ReservationPrice { get; set; }
    public DateTimeOffset StartTime { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset EndTime { get; set; }
    public DateTimeOffset? InactivatedAt { get; set; }

    public bool IsOpen => EndTime > DateTimeOffset.UtcNow && InactivatedAt == null;
    public bool IsClosed => EndTime <= DateTimeOffset.UtcNow;
    public bool IsActive => InactivatedAt == null;

    public int UserId { get; set; }
    public User User { get; set; } = null!;

    public List<Bid> Bids { get; set; } = [];
}

File: frontend/src/types/auction.ts
import type { BidSummaryResponse } from "./bid";

export type AuctionStatus = "open" | "closed" | "all";

export interface AuctionSummaryResponse {
  auctionId: number;
  title: string;
  startingPrice: number;
  currentHighestBid: number | null;
  startTime: string;
  endTime: string;
  ownerUsername: string;
  bidCount: number;
  imageUrl: string | null;
}

export interface AuctionDetailResponse {
  auctionId: number;
  title: string;
  description: string;
  ownerUsername: string;
  ownerId: number;
  startingPrice: number;
  currentHighestBid: number | null;
  startTime: string;
  endTime: string;
  isOpen: boolean;
  imageUrl: string | null;
  bids: BidSummaryResponse[];
}

export interface CreateAuctionRequest {
  title: string;
  description: string;
  startingPrice: number;
  reservationPrice?: number;
  endTime: string;
  imageUrl?: string;
}

export interface UpdateAuctionRequest {
  title: string;
  description: string;
  imageUrl?: string;
}
